prober/http.go

```diff
@@ -397,10 +400,27 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 	request = request.WithContext(ctx)
 
 	for key, value := range httpConfig.Headers {
-		if strings.Title(key) == "Host" {
+		normalizedKey := strings.Title(key)
+
+		if normalizedKey == "Host" {
 			request.Host = value
 			continue
 		}
+
+		// If there's a compression setting, and there's also an
+		// accept-encoding header, they MUST match, otherwise we
+		// end up requesting something that doesn't include the
+		// specified compression, and that's likely to fail,
+		// depending on how the server is configured. Testing
+		// that the server _ignores_ Accept-Encoding, e.g. by
+		// not including a particular compression in the header
+		// but expecting it in the response falls out of the
+		// scope of the tests we perform.
+		if httpConfig.Compression != "" && normalizedKey == "Accept-Encoding" && !isEncodingAcceptable(httpConfig.Compression, value) {
```

Comment:
Can we check that at unmarshalling?

Reply:
Sure! I have moved some of the code around to check the config is valid at unmarshalling.